homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -218,57 +221,153 @@ def _async_register_bridge_device(
 
 
 @callback
-def _async_register_button_devices(
+def _async_setup_keypads(
     hass: HomeAssistant,
     config_entry_id: str,
     bridge: Smartbridge,
     bridge_device: dict[str, Any],
-    button_devices_by_id: dict[int, dict],
-) -> tuple[dict[str, dict], dict[int, DeviceInfo]]:
-    """Register button devices (Pico Remotes) in the device registry."""
+) -> LutronKeypadData:
+    """Register keypad devices (Keypads and Pico Remotes) in the device registry."""
+
     device_registry = dr.async_get(hass)
-    button_devices_by_dr_id: dict[str, dict] = {}
-    device_info_by_device_id: dict[int, DeviceInfo] = {}
-    seen: set[str] = set()
+
     bridge_devices = bridge.get_devices()
+    bridge_buttons = bridge.buttons
+
+    dr_device_id_to_keypad: dict[str, LutronKeypad] = {}
+    keypads: dict[int, LutronKeypad] = {}
+    keypad_buttons: dict[int, LutronButton] = {}
+    keypad_button_names_to_leap: dict[int, dict[str, int]] = {}
+
+    keypad: LutronKeypad
+    for bridge_button in bridge_buttons.values():
+
```

Comment:
Should keypad be scoped inside the loop here?

Are re-using a variable in the next loop that we are defining in the loop?

Reply:
I don't want keypad changing/resetting on each run of the loop.  Only when the parent_device is different

Reply:
Looping through all the buttons:
- if new keypad, record the keypad details
- either way process the button

it's discovering the keypads and buttons at the same time.

Reply:
Line 242 is just for Typing.  I don't need this line I guess.

Reply:
addressed with commit https://github.com/home-assistant/core/pull/80797/commits/f787a1f4e10048ac2c59c63a35a581f5cfee0934